### PR TITLE
"Added_synonym_application_timeout_unit_for_run"

### PR DESCRIPTION
**Pull request by vinu2003**
## Please provide the following details to expedite Pull Request review:

----

## Description of change
Why is this change needed?
Options for run commands like application,timeout,unit can be abbreviated for ease of use in command line

## QA steps
How do we verify that the change works?
Configured LXD, created controller and deployed simple application mediawiki.
Verified after making changes to the code using juju CLI - run command with combinations.
for eg: juju run --app=mysql -u mysql/0  "ls"
## Documentation changes
Does it affect current user workflow? CLI? API?
N/A
## Bug reference
Does this change fix a bug? Please add a link to it
https://bugs.launchpad.net/juju/+bug/1751321.

### CODE REVIEW

**Comment by vinu2003** [comment]
$$merge$$

File: cmd/juju/commands/run_test.go [hunk 1]
```diff
@@ -151,6 +171,10 @@ func (*RunSuite) TestTimeoutArgParsing(c *gc.C) {
 		message: "two hours",
 		args:    []string{"--timeout=2h", "--all", "sudo reboot"},
 		timeout: 2 * time.Hour,
+	}, {
+		message: "5 minutes",
+		args:    []string{"-t","5m", "--all", "sudo reboot"},
```
Comment (wallyworld) [inline cmd/juju/commands/run_test.go]: Looks like a few whitespace issues, did you run go fmt ./... before pushing?
Ideally you'd have the git precommit hook set up to do this and also run go vet

File: cmd/juju/commands/run.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ are able to use this command.
 Targets are specified using either machine ids, application names or unit
 names.  At least one target specifier is needed.
 
-Multiple values can be set for --machine, --application, and --unit by using
+Multiple values can be set for --machine, --application/--app/-a, and --unit/-u by using
```
Comment (wallyworld) [inline cmd/juju/commands/run.go]: I wouldn't complicate the comment here. Keep all this as --application --unit etc
Just add a new, standalone paragraph which says that --application can be shorted to --app or -a, --unit to -u etc

File: cmd/juju/commands/run.go [hunk 3]
```diff
@@ -63,8 +63,11 @@ If the target is an application, the command is run on all units for that
 application. For example, if there was an application "mysql" and that application
 had two units, "mysql/0" and "mysql/1", then
   --application mysql
+  --app mysql
```
Comment (wallyworld) [inline cmd/juju/commands/run.go]: As per the above comment, delete the lines here and just below

File: cmd/juju/commands/run.go [hunk 4]
```diff
@@ -102,9 +105,13 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 	})
 	f.BoolVar(&c.all, "all", false, "Run the commands on all the machines")
 	f.DurationVar(&c.timeout, "timeout", 5*time.Minute, "How long to wait before the remote command is considered to have failed")
+	f.DurationVar(&c.timeout, "t", 5*time.Minute, "How long to wait before the remote command is considered to have failed")
```
Comment (wallyworld) [inline cmd/juju/commands/run.go]: The code would be better like this:

for _, flag := range []string{"timeout", "t"} {
    f.DurationVar(&c.timeout, flag, 5*time.Minute, "How long to wait before the remote command is considered to have failed")
}

etc (similar for unit and application)

File: cmd/juju/commands/run.go [hunk 5]
```diff
@@ -134,7 +141,7 @@ func (c *runCommand) Init(args []string) error {
 		}
 	} else {
 		if len(c.machines) == 0 && len(c.services) == 0 && len(c.units) == 0 {
-			return errors.Errorf("You must specify a target, either through --all, --machine, --application or --unit")
+			return errors.Errorf("You must specify a target, either through --all, --machine, --application/--app/-a or --unit/-u")
```
Comment (wallyworld) [inline cmd/juju/commands/run.go]: Just keep these as --application etc

**Comment by vinu2003** [comment]
The last commit has conflicts

**Pull request closed without merging**